### PR TITLE
XRechnung fixes

### PR DESCRIPTION
**Pull request by urakozz**
Hey guys, I'm happy to contribute based on the feedback from our production tests.

New API looks fancy, big kudos and thank you!

We have identified several small issues that I addressed. I use this tool for the validation and it went nuts after the recent update. However, I managed to find new fields and fixed a couple of missing ones

- Delivery date was missing `format`
- Header note was missing `Type`
- It was impossible to create `Unspecified` payment method (without a specific IBAN/Card/DirectDebit)
- Invoice number didn't keep number without `Series`

BTW I use this tool to validate xml https://www.portinvoice.com

For some reason test regeneration didn't work for me, could you please regenerate it after addressing those updates? 🙏

### CODE REVIEW

File: internal/gtoc/header.go [hunk 1]
```diff
@@ -29,7 +29,8 @@ func (c *Converter) newHeader(inv *bill.Invoice) error {
 		notes := make([]*document.Note, 0, len(inv.Notes))
 		for _, n := range inv.Notes {
 			notes = append(notes, &document.Note{
-				Content: n.Text,
+				Content:     n.Text,
+				SubjectCode: n.Code,
```
Comment (apardods) [inline internal/gtoc/header.go:33]: I would add a nil check here, as it is possible for a note not to include Code.

**Review by apardods — CHANGES_REQUESTED** [review]
Thank you very much for your help! Looks fantastic!

File: internal/gtoc/header.go (hunk 1 above)
Comment (urakozz) [reply internal/gtoc/header.go:33]: @apardods Hey, thx for having a look. I just checked the code, and both entities `cbc.Note` and `document.Note` have fields `Code` and `SubjectCode` respectively declared as just a `string`, so they can't be nil.

I tested note without a Code, it works as an empty string and doesn't end up in the final xml

Comment (apardods) [reply internal/gtoc/header.go:33]: Perfect! As long it an invoice with no code does not show the empty field great

**Review by apardods — APPROVED** [review]

**Comment by apardods** [comment]
Seems like @samlown needs to approve the test run for CodeQL. I also suggest including the .github folder in #5 to include the lint test.

**Comment by urakozz** [comment]
@apardods thx for the approval, I fixed a small lint issue

**Comment by apardods** [comment]
> @apardods thx for the approval, I fixed a small lint issue

Thank you for your help! I just have to figure out how to run the CodeQL action to merge, although it should be running automatically...

**Comment by samlown** [comment]
@urakozz @apardods many thanks for this! 👍  Merged!

**Comment by urakozz** [comment]
@samlown @apardods thank you too, party time 🎉

**Pull request merged**